src/core/lib/event_engine/ares_resolver.cc

```
@@ -0,0 +1,714 @@
+  if (grpc_parse_ipv4_hostport(hostport.c_str(), &addr,
+                               false /* log errors */) ||
+      grpc_parse_ipv6_hostport(hostport.c_str(), &addr,
+                               false /* log errors */)) {
+    // Early out if the target is an ipv4 or ipv6 literal.
+    std::vector<EventEngine::ResolvedAddress> result;
+    result.emplace_back(reinterpret_cast<sockaddr*>(addr.addr), addr.len);
+    event_engine_->Run(
+        [callback = std::move(callback), result = std::move(result)]() mutable {
+          callback(std::move(result));
+        });
+    return;
+  }
+  grpc_core::MutexLock lock(&mutex_);
+  callback_map_.emplace(++id_, std::move(callback));
+  auto* resolver_arg = new HostnameQueryArg(this, id_, name, port);
+  if (IsIpv6LoopbackAvailable()) {
+    ares_gethostbyname(channel_, std::string(host).c_str(), AF_UNSPEC,
+                       &AresResolver::OnHostbynameDoneLocked,
+                       static_cast<void*>(resolver_arg));
+  } else {
+    ares_gethostbyname(channel_, std::string(host).c_str(), AF_INET,
+                       &AresResolver::OnHostbynameDoneLocked,
+                       static_cast<void*>(resolver_arg));
+  }
+  CheckSocketsLocked();
+  MaybeStartTimerLocked();
+}
+
+void AresResolver::LookupSRV(
+    absl::string_view name,
+    EventEngine::DNSResolver::LookupSRVCallback callback) {
+  absl::string_view host;
+  absl::string_view port;
+  if (!grpc_core::SplitHostPort(name, &host, &port)) {
+    event_engine_->Run(
+        [callback = std::move(callback),
+         status = absl::InvalidArgumentError(absl::StrCat(
+             "Unparseable name: ", name))]() mutable { callback(status); });
+    return;
+  }
+  GPR_ASSERT(!host.empty());
+  // Don't query for SRV records if the target is "localhost"
+  if (absl::EqualsIgnoreCase(host, "localhost")) {
+    event_engine_->Run(
+        [callback = std::move(callback),
+         status = absl::UnknownError(
```

Do we want to return an error here, or do we want to just return an empty result?  I don't remember what the behavior of the old code was.

---

You are right. The old code reports an Ok status with an empty address list. Changed it accordingly.

https://source.corp.google.com/piper///depot/google3/third_party/grpc/src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc;l=1091-1096;bpv=1;bpt=1